internal/validate/kube/kube.go

```diff
@@ -50,41 +58,86 @@ func Quiet() Option {
 	}
 }
 
-type validation func(ctx context.Context, config *Config) ([]validate.Result, error)
+func Eks() Option {
+	eksConfig, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		log.Printf("error while loading config: %s\n", err)
+	}
+
+	ec2Config, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		log.Printf("error while loading config: %s\n", err)
+	}
+
+	iamConfig, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		log.Printf("error while loading config: %s\n", err)
+	}
+
```

Comment:
I would move these to a separate function in `eks.go`, then call that function in the nested `if cfg.eks` block below. This way you can use the context already passed to `Validate` instead of the `TODO` context. Since `config` is passed around as a reference, you should still have no issue storing the configs for eks, ec2, and iam in the `Config` struct.

Reply:
Done!